projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-trait.php

```diff
@@ -26,18 +26,162 @@ trait Jetpack_WooCommerce_Analytics_Trait {
 	 */
 	protected $cart_checkout_templates_in_use;
 
+	/**
+	 * The content of the cart page or where the cart page is ultimately derived from if using a template.
+	 *
+	 * @var string
+	 */
+	protected $cart_content_source = '';
+
+	/**
+	 * The content of the checkout page or where the cart page is ultimately derived from if using a template.
+	 *
+	 * @var string
+	 */
+	protected $checkout_content_source = '';
+
+	/**
+	 * Gets the content of the cart/checkout page or where the cart/checkout page is ultimately derived from if using a template.
+	 * This method sets the class properties $checkout_content_source and $cart_content_source.
+	 *
+	 * @return void Does not return, but sets class properties.
+	 */
+	public function find_cart_checkout_content_sources() {
+
+		/**
+		 * The steps we take to find the content are:
+		 * 1. Check the transient, if that contains content and is not expired, return that.
+		 * 2. Check if the cart/checkout templates are in use. If *not in use*, get the content from the pages and
+		 *    return it, there is no need to dig further.
+		 * 3. If the templates *are* in use, check if the `page-content-wrapper` block is in use. If so, get the content
+		 *    from the pages (same as step 2) and return it.
+		 * 4. If the templates are in use but `page-content-wrapper` is not, then get the content directly from the
+		 *    template and return it.
+		 * 5. At the end of each step, assign the found content to the relevant class properties and save them in a
+		 *    transient with a 1-day lifespan. This will prevent us from having to do this work on every page load.
+		 */
+
+		$cart_checkout_content_cache_transient_name = 'jetpack_woocommerce_analytics_cart_checkout_content_sources';
+
+		$transient_value = get_transient( $cart_checkout_content_cache_transient_name );
+
+		if (
+			false !== $transient_value &&
+			! empty( $transient_value['checkout_content_source'] ) &&
+			! empty( $transient_value['cart_content_source'] )
+		) {
+			$this->cart_content_source     = $transient_value['cart_content_source'];
+			$this->checkout_content_source = $transient_value['checkout_content_source'];
+			return;
+		}
+
+		// Cart/Checkout *pages* are in use if the templates are not in use. Return their content and do nothing else.
+		if ( ! $this->cart_checkout_templates_in_use ) {
+			$cart_page_id     = wc_get_page_id( 'cart' );
+			$checkout_page_id = wc_get_page_id( 'checkout' );
+			$cart_page        = get_post( $cart_page_id );
+			$checkout_page    = get_post( $checkout_page_id );
+
+			if ( $cart_page && isset( $cart_page->post_content ) ) {
+				$this->cart_content_source = $cart_page->post_content;
+			}
+			if ( $checkout_page && isset( $checkout_page->post_content ) ) {
+				$this->checkout_content_source = $checkout_page->post_content;
+			}
```

Comment:
Nitpic: I'd add a line break here to increase the readability.

```suggestion
			if ( $cart_page && isset( $cart_page->post_content ) ) {
				$this->cart_content_source = $cart_page->post_content;
			}
			
			if ( $checkout_page && isset( $checkout_page->post_content ) ) {
				$this->checkout_content_source = $checkout_page->post_content;
			}
```